sympy/core/mod.py

```diff
@@ -117,15 +117,17 @@ def doit(p, q):
                 net = Add(*non_mod_l) + Add(*[i.args[0] for i in mod_l])
                 return cls(net, q)
 
-        elif isinstance(p, Mul):
+        if isinstance(p, Mul):
             # separating into modulus and non modulus
             both_l = non_mod_l, mod_l = [], []
             for arg in p.args:
                 both_l[isinstance(arg, cls)].append(arg)
 
-            if mod_l and all(inner.args[1] == q for inner in mod_l):
+            was = non_mod_l[:]
```

Comment:
Hi @smichr 
I just updated the pull request after including the changes suggested above.